### PR TITLE
Console (Scala REPL) API port from sbt-pamflet

### PR DESCRIPTION
**Pull request by knirski**
Implements #170 . Added some additional tests as suggested.

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
Hi @sepulkarz,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by eed3si9n** [comment]
Nice.

**Comment by eed3si9n** [comment]
@sepulkarz Could you let us know when you sign the CLA plz?

File: internal/compiler-bridge/src/main/java/xsbti/ConsoleFactory.java
Comment (jvican) [inline internal/compiler-bridge/src/main/java/xsbti/ConsoleFactory.java]: Small suggestion: Can we remove these comments? Not useful anymore, sbt-pamflet will probably go away in next sbt 1.0 :smile:.

File: internal/compiler-bridge/src/main/scala/xsbt/Console.scala
Comment (jvican) [inline internal/compiler-bridge/src/main/scala/xsbt/Console.scala]: We can remove all these comments, we won't need to do any magic with class loaders in Zinc.

File: internal/compiler-bridge/src/main/scala/xsbt/Console.scala
Comment (jvican) [inline internal/compiler-bridge/src/main/scala/xsbt/Console.scala]: We can remove this tiny compat bit, we don't support Scala 2.9.x anymore :wink:.

File: internal/compiler-bridge/src/test/scala/xsbt/REPLSpecification.scala
Comment (jvican) [inline internal/compiler-bridge/src/test/scala/xsbt/REPLSpecification.scala]: This test suite looks amazing.

**Review by jvican — APPROVED** [review]
Hey @sepulkarz, great changes and great testing suite! We really wanted to get this done soon, we appreciate your effort into making it a reality.
  
I've left some very small suggestions. Can you address them and sign the CLA now? @eed3si9n Do you have further comments on this one? I'll leave it to you to approve and merge it.

**Comment by jvican** [comment]
@dwijnand Can you have a look at this too?

**Comment by eed3si9n** [comment]
We should put the following header on all files:

```scala
/*
 * Zinc - The incremental compiler for Scala.
 * Copyright 2011 - 2017, Lightbend, Inc.
 * Copyright 2008 - 2010, Mark Harrah
 * This software is released under the terms written in LICENSE.
 */
```

For port from sbt-pamflet, we should keep the original commit if possible with the `--author` set to the original author (me).

**Comment by knirski** [comment]
Thanks for the comments. I will try to adapt the PR according to them.

**Comment by knirski** [comment]
@eed3si9n Some of the interface files were touched by more than one commit. Also not all of the files were copied, as `Logger` and `F0` are already present in `util-interface` dependency. Shall I cherry-pick the original 3 commits and delete `Logger` and `F0` in a separate commit?

**Comment by eed3si9n** [comment]
Yea. I think that would make sense. In this case it's me, so it's fine, but we have to be diligent about crediting the original authors in OSS partly for legal reasons.

**Comment by knirski** [comment]
Understood. A quick question: CLA mentions "sign and return this Agreement to the Company below by using GitHub", but the page that links to it (http://www.lightbend.com/contribute/cla) says "You do not need to provide Lightbend the signed agreement.". I'm not sure how to proceed.

**Comment by eed3si9n** [comment]
You just have to verify the CLA via http://www.lightbend.com/contribute/cla/verify once it's signed, instead of physically mailing in the copy to us.

**Comment by jvican** [comment]
@eed3si9n I think CLA is already signed, otherwise CI wouldn't have kicked in. Can you approve and merge?

**Comment by knirski** [comment]
@eed3si9n Thanks, I did that. @jvican I think it's not ready to merge yet - I want to do some corrections you guys suggested. I will let you know shortly.

**Comment by romanowski** [comment]
@sepulkarz great work! LGTM and waiting for mentioned 'small corrections'  :)

**Comment by knirski** [comment]
I made all the suggested changes :). One last tidbit - it seems like the existing test suites don't have a copyright header, so I didn't put one in the new test suite.

**Comment by jvican** [comment]
@sepulkarz I think it's good, don't worry. Another PR should target putting a copyright header to those test source files :wink:.
  
@eed3si9n Merge when you agree with the changes. I've already approved this PR.
  
Thanks again Krzystoff! Hope to see you around :smile:.

**Review by eed3si9n — APPROVED** [review]
Thanks for addressing the review comments.

**Comment by eed3si9n** [comment]
LGTM pending Travis.

**Comment by jvican** [comment]
Merging, Travis is happy.

**Pull request merged**
